RUFAS/input_manager.py

```diff
@@ -91,3 +91,78 @@ def _load_data(self) -> None:
                                    info_map)
             except Exception as e:
                 raise e
+
+    def _validate_data(self, eager_termination: bool = True) -> bool:
+        """
+        Validates input data.
+
+        Args
+        ----
+        eager_termination : bool, optional
+            Flag to determine if the process should terminate upon finding invalid data.
+
+        Returns
+        -------
+        bool
+            True if all data is valid; False otherwise.
+        """
+        info_map = {"class": self.__class__.__name__,
+                    "function": self._validate_data.__name__,
+                    }
+        for key, value in self.__pool.items():
+            if not self._validate(key, value):
+                if eager_termination:
+                    om.add_error("Invalid data.", f"Invalid data found: {key} - {value}", info_map)
+                    if not self._fix_data(key, value):
+                        om.add_error("Data not fixable.",
+                                     f"Unable to fix the invalid data: {key} - {value}. Terminating the process.",
+                                     info_map)
+                        return False
+                    else:
+                        om.add_warning("Data was fixable.",
+                                       f"Invalid data found: {key} - {value} but was fixable.",
+                                       info_map)
```

Comment:
Even if `eager_termination` is set to False, I think it would still be good to raise a warning and/or attempt to fix any invalid data found. The difference should be whether or not the the function exits when invalid/unfixable data is found. 

However if this is not done, then it would be better to return True at the very beginning of the function when `eager_termination` is False, because it won't matter if any of the data is invalid so iterating through the pool could be entirely skipped.

Reply:
What about something like this?
```suggestion
       for key, value in self.__pool.items():
            if not self._validate(key, value):
                om.add_error("Invalid data.", f"Invalid data found: {key} - {value}", info_map)
                if not self._fix_data(key, value) and eager_termination:
                    om.add_error("Data not fixable.",
                                 f"Unable to fix the invalid data: {key} - {value}. Terminating the process.",
                                 info_map)
                    return False
                elif not self._fix_data(key, value):
                    om.add_error("Data not fixable.",
                                 f"Unable to fix invalid data: {key} - {value}. Continuing the process.",
                                 info_map)
                else:
                    om.add_warning("Data was fixable.",
                                   f"Invalid data found: {key} - {value} but was fixable.",
                                   info_map)
        return True
```

Reply:
That looks good, I suggest running `_fix_data()` once and saving the result in variable though